src/services/NumberService.ts

```diff
@@ -55,15 +55,64 @@ export class TestNumberService {
     }
 
     /**
-     * Retrieves the last test number
+     * Creates a new trailer id in the database.
+     * @param attempt - the current number of attempts for generating a TrailerId.
+     * @param awsError - AWS error to be passed when the request fails otherwise null.
      */
-    public getLastTestNumber(): Promise<TestNumber> {
+    public createTrailerId(attempts: number, awsError: AWSError | null): Promise<TrailerId> {
```

Comment:
There's so much similarity between these two functions, I want to extract it to a function, and then just pass in the things that make them different

Reply:
fixed